src/librustc_resolve/lib.rs

```diff
@@ -2474,6 +2482,19 @@ impl<'a, 'crateloader: 'a> Resolver<'a, 'crateloader> {
         self.ribs[TypeNS].pop();
     }
 
+    fn resolve_type_definition_with_self_ty(&mut self,
```

Comment:
Is there any reason not to inline this (with an explanatory comment)?

Reply:
It's cleaner and clearer this way, I think.

Reply:
I disagree. You could have only one `with_type_parameter_rib` (exactly where it used to be), and then only make the `with_self_rib` conditional.

Reply:
Fair enough, I'll do that then. Just to make you happy!